src/integrations/front-end/open-graph-oembed.php

```diff
@@ -68,112 +85,64 @@ public function __construct( Meta_Helper $meta, Image_Helper $image ) {
 	 * @param array    $data The oEmbed data.
 	 * @param \WP_Post $post The current Post object.
 	 *
+	 * @return array $filter_data - An array of oEmbed data with modified values where appropriate.
 	 * @link https://developer.wordpress.org/reference/hooks/oembed_response_data/ for hook info.
 	 *
-	 * @return array $filter_data - An array of oEmbed data with modified values where appropriate.
 	 */
 	public function set_oembed_data( $data, $post ) {
 		// Data to be returned.
-		$filter_data = $data;
+		$this->data    = $data;
+		$this->post_id = $post->ID;
 
-		$filter_data = $this->set_title( $filter_data, $post->ID );
-		$filter_data = $this->set_image( $filter_data, $post->ID );
+		$this->set_title();
+		$this->set_description();
+		$this->set_image();
 
-		return $filter_data;
+		return $this->data;
 	}
 
 	/**
-	 * Sets the title if it has been configured.
-	 *
-	 * @param array $data    The data.
-	 * @param int   $post_id The post id.
-	 *
-	 * @return array The modified data array.
+	 * Sets the OpenGraph title if configured.
 	 */
-	protected function set_title( array $data, $post_id ) {
-		$opengraph_title = $this->meta->get_value( 'opengraph-title', $post_id );
+	protected function set_title() {
+		$opengraph_title = $this->meta->for_post( $this->post_id )->open_graph_title;
```

Comment:
Have you considered storing the value of `$this->meta->for_post( $this->post_id )` in a class property?

Reply:
`open_graph_title` and `open_graph_description` use a magic getter, don't know if that'll work, this seems more readable to me too :)

Reply:
Okay, I'm fine with that 👍